harness/determined/cli/command.py

```diff
@@ -282,15 +284,38 @@ def config(args: Namespace) -> None:
     print(render.format_object_as_yaml(res_json["config"]))
 
 
-def _set_nested_config(config: Dict[str, Any], key_path: List[str], value: Any) -> Dict[str, Any]:
-    current = config
-    for key in key_path[:-1]:
-        current = current.setdefault(key, {})
-    current[key_path[-1]] = value
-    return config
-
+# Convert config overrides in dot notation to json path
+# For example the caller of this func will break down the config override
+#   --config slurm.sbatch_args=[--time=05:00]
+# into key slurm.sbatch_args, value [--time=05:00]
+# And this func will return string in json form:
+# 'slurm': {'sbatch_args': ['--time=05:00']}
+def _dot_to_json(key: Any, value: Any) -> Any:
+    json_output: Any = {}
+    path = key.split(".")
+    target = reduce(lambda d, k: d.setdefault(k, {}), path[:-1], json_output)
+    target[path[-1]] = value
+    return json_output
+
+
+# A recursive function to replace any value val in a Dict with a new value rval
+def _replace_value(data_dict: Dict[str, Any], val: Any, rval: Any) -> None:
+    for key in data_dict.keys():
+        if data_dict[key] == val:
+            data_dict[key] = rval
+        elif type(data_dict[key]) is dict:
+            _replace_value(data_dict[key], val, rval)
+
+
+def parse_config_overrides(config: Dict[str, Any], overrides: Iterable[str]) -> Dict[str, Any]:
+    if any(True for _ in overrides):
+        # Some key may have value None of NoneType and will raise TypeError if used.
+        # For example:
+        # TypeError: argument of type 'NoneType' is not iterable
+        # TypeError: 'NoneType' object does not support item assignment
+        # Convert NoneType value None in config to empty string {}
+        _replace_value(config, None, {})
```

Comment:
I am worried about this solution because because it replaces *all* None values in the config with an empty dict.
We have some fields in the config such as `bind_mounts` which are either an List or None. replacing a none with a dictionary will likely cause a typing error during validation / parsing.

Plus, this effectively shuts off any opportunity for a user to explicitly set anything to None. This is already difficult to do today because the golang side parsing can't differentiate missing values from nulls, but we shouldn't make it worse.

Can we please make `None`s into dicts only when we are trying to apply an override?

Reply:
Got you. Will do it

Reply:
fixed